content/docs/migration_1.0/Rust.md

```diff
@@ -0,0 +1,380 @@
+            );
+        }
+        // Difference 3: ReplyError instead of Value
+        Err(err) => {
+            println!("{}", err.payload().deserialize::<String>().unwrap());
+        }
+    }
+}
+```
+
+We have also added the ability to get underlying Handlers from `Queryables`, a
+
+```rust
+let queryable = session
+    .declare_queryable(&key_expr)
+    .await
+    .unwrap();
+
+let handler: &Receiver<Query> = queryable.handler();
+// or mutable handler
+let mut_handler:&mut Receiver<Query> = queryable.handler_mut();
+```
+
+
+## Use accessors to get private members
+
+We encapsulate members of structs, and they can’t be accessed directly now. 
+The only way to access Struct values is to use the getter function associated with them. 
+Let’s take the subscriber as an example here.
+
+- Zenoh 0.11.x
+
+```rust
+while let Ok(sample) = subscriber.recv_async().await {
+    println!(
+        ">> [Subscriber] Received {} ('{}': '{}')",
+        sample.kind,
+        sample.key_expr.as_str(),
+        sample.value
+    );
+}
+```
+
+- Zenoh 1.0.0
+
+```rust
+while let Ok(sample) = subscriber.recv_async().await {
+    println!(
+        ">> [Subscriber] Received {} ('{}': '{:?}')",
+        sample.kind(),
+        sample.key_expr().as_str(),
+        sample.payload()  // Ignore the deserialization
+    );
+}
+```
+
+## Support RingChannel to receive data
+
+Besides using a callback to receive data, we can also receive the data from a default FIFO channel. However, sometimes we only care about the latest data and want to discard the oldest data. 
+We can use `RingChannel` to get this behaviour.
+You can take a look at the complete code in `examples/examples/z_pull.rs`.
+
+```rust
+let subscriber = session
+    .declare_subscriber(&key_expr)
+    .with(RingChannel::new(size))
+    .await
+    .unwrap();
+```
+
+⚠️ Note:  We **no longer** support **Pull** mode in Zenoh
+
+To get the same behavior of a Zenoh 0.11.0 `PullSubscriber`, please make use of a `RingChannel` an example of this is illustrated in `z_pull.rs`.
+
+## Timestamps
+
+We now tie generating a timestamp to a Zenoh session, with the timestamp inheriting the `ZenohID` of the session.
+
+Note that a Zenoh session will only be able to generate a timestamp if the `timestamping` configuration option is enabled.
+
+- Zenoh 0.11.x
+
+```rust
+let timestamp : Timestamp =  zenoh::time::new_reception_timestamp();
+```
+
+- Zenoh 1.0.0
+
+```rust
+let session: Session = zenoh::open();
+// If the `timestamping` configuration is disabled, this call will return `None`.
+let timestamp: Option<Timestamp> = session::new_timestamp();
+```
+
+This will affect user-created plugins and applications that need to generate timestamps.
+
+## Feature Flags
+
+Removed:
+
+- `complete_n`: due to a Legacy code cleanup
+
+## Storages
+
+<aside>
+⚠️ The `storage-manager` will fail to launch if the `timestamping` configuration option is disabled.
+
+</aside>
```

Comment:
Out of curiosity, do we know if the `<aside>` work?

Reply:
Running locally, aside does not work.
I will remove it in the next commit.
Good spot.